libs/exavmlib/lib/Code.ex

```diff
@@ -23,22 +23,20 @@ defmodule Code do
   @moduledoc """
   This module is to satisfy certain code loading checks in Elixir,
   specifically with regards to protocols support.
-  The functions are noop, and doesn't perform the actual checks,
-  as they are not relevant on AtomVM.
   """
 
   @doc """
   required for protocols to work with Elixir >= 1.16, due to code loading checks.
   """
   def ensure_compiled(module) do
-    {:module, module}
+    :code.ensure_loaded(module)
   end
 
   @doc """
   previously required for protocols support, due to code loading checks.
   """
   @deprecated "Use Code.ensure_compiled/1 instead"
-  def ensure_compiled?(_) do
-    true
+  def ensure_compiled?(module) do
+    :code.ensure_loaded(module)
```

Comment:
this should return bool right? 

eg https://github.com/elixir-lang/elixir/blob/0fb34f922f7d0d3c01dba475a140a89876d326ea/lib/elixir/lib/code.ex#L2036